portal-api/src/external-api/bank-holidays.js

```diff
@@ -0,0 +1,34 @@
+const axios = require('axios');
+const dotenv = require('dotenv');
+
+dotenv.config();
+const { EXTERNAL_API_URL, EXTERNAL_API_KEY } = process.env;
+
+const headers = {
+  'Content-Type': 'application/json',
+  'x-api-key': EXTERNAL_API_KEY,
+};
+
+/**
+ * Fetches details of UK bank holidays from external-api.
+ * @returns {Promise<object>}
+ */
+const getBankHolidays = async () => {
+  const response = await axios.get(`${EXTERNAL_API_URL}/bank-holidays`, { headers });
+  return response.data;
+};
+
+/**
+ * Fetches details of UK bank holidays for the specified region from external-api.
+ * @param {'england-and-wales' | 'scotland' | 'northern-ireland'} region
+ * @returns {Promise<Date[]>}
+ */
+const getBankHolidaysForRegion = async (region) => {
+  const allUkBankHolidays = await getBankHolidays();
+  return allUkBankHolidays[region].events.map((event) => new Date(event.date));
```

Comment:
Caveat that I know this isn't all your work but it feels weird that our getBankHolidays and getBankHolidaysForRegion return data in a different format

Reply:
Have updated the description and function name to make this a bit clearer. I could instead just return `allUkBankHolidays[region]` here to make this more consistent, but my assumption was that we'd always want to do this mapping to just get the dates.